recipes/qr-code-generator/all/CMakeLists.txt

```diff
@@ -24,19 +24,28 @@ if(WIN32 AND BUILD_SHARED_LIBS)
     set(CMAKE_VS_INCLUDE_INSTALL_TO_DEFAULT_BUILD TRUE)
 endif()
 
-set(SRC
-    source_subfolder/cpp/QrCode.cpp
-)
+if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/source_subfolder/cpp/QrSegment.cpp)
```

Comment:
Won't it make more sense to move this to patches?
e.g. let each version create a `files.cmake` file that defines these variables and include that here.